src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -665,7 +751,39 @@ public static String Join<T>(String separator, IEnumerable<T> values)
             }
         }
 
-        // Joins an array of strings together as one string with a separator between each original string.
+        // Joins a string IEnumerable together as one string with a char separator between each original string.
+        //
+        [ComVisible(false)]
+        public static String Join(Char separator, IEnumerable<String> values) {
```

Comment:
@terrajobst @weshaggard as you guys reviewed these APIs do you have any second thoughts about this?